doc/console/serial-console-HLD.md

```diff
@@ -0,0 +1,214 @@
+
+###  1.1. <a name='Scope'></a>Scope  
+
+This hld doc for serial cli global configurations describes the requirements, architecture and general flow details of serial connection config in SONIC OS based switches.
+
+###  1.2. <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations 
+
+	serial - secure shell
+	TCP - Transmission Control protocol	
+	
+###  1.3. <a name='Overview'></a>Overview 
+
+We want to allow configuring serial console global settings. This will feature will include two configurations on first phase.
+
+###  1.4. <a name='Requirements'></a>Requirements
+
+This feature requires a dedicated table in the configuration DB, enhancements of hostcfg demon and separate service that updates configutaion files. In order to update the serial-console configurations, we need to have write access in /etc/profile.d/, /proc/sys/kernel/sysrq and /etc/sysctl.d/95-sysrq-sysctl.conf.
+
+##  2. <a name='ArchitectureDesign'></a>Architecture Design 
+###  2.1. <a name='ConfigModules'></a>Configuration modules
+![serial_console_config](serial_console_config.png)
+
+We want to enhance configDB to include table for serial-console global configurations. In addition, serial_console.service will be added to modify configuration files.
+
+##  3. <a name='High-LevelDesign'></a>High-Level Design 
+
+We want to enable serial-console configuration in SONIC. In order to do so will touch few areas in the system:
+1. configDB - to include a dedicated table for configurations
+2. hostcfg demon - to trigger dedicated service on config apply.
+3. OS config files - specific for this stage we are only /etc/profile.d/serial_tmout.sh and /etc/sysctl.d/95-sysctl.d/95-sysrq-sysctl.conf and /proc/sys/kernel/sysrq are going to be modifed by the service-script.
+
+##### Flow diagram
+![serial_console_flow](serial_console_flow.png)
+### 3.1 <a name='Flow description'></a>Flow description
+When the feature is enabled, by modifying the DB manually, user will set serial-console configurations by modifing CONFIG_DB in SERIAL_CONSOLE table.
+
+The hostcfgd daemon will be extended to listen to confogurations from SERIAL_CONSOLE table and restarts the serial_console.service. Serial console script will read SERIAL_CONSOLE table and update config files accordingly.
+
+
+### 3.2 <a name='serial-console configurations'></a>serial-console configurations
+
+We want to enable configuring the following attributes:
+| Policy             | Action                                                                | Param values        | Default      |
+|--------------------|-----------------------------------------------------------------------|---------------------|--------------|
+| auto logout        | Inactivity timeout for serial-console session                         | 0-35000 (min)       | 15           |
+| sysrq capabilities | Enabling or disabling SysRq functionality for serial-consoles         | enabled/disabled    | disabled     |
```

Comment:
i saw some test are using sysrq in sonic-mgmt.

tests/scripts/sai_qualify/sai_warmboot.sh:        echo 1 > /proc/sys/kernel/sysrq

can we check if this won't affect the tst?

@vaibhavhd  for visibility.